tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -287,6 +289,129 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
+		framework.ExpectNoError(err)
+		restoreSecret := fixture.GetBackupSecret(tcTo, "")
+		_, err = c.CoreV1().Secrets(ns).Create(restoreSecret)
+		framework.ExpectNoError(err)
+		cred := credentials.NewSharedCredentials("", "default")
+		val, err := cred.Get()
+		framework.ExpectNoError(err)
+		backupS3Secret := fixture.GetS3Secret(tcFrom, val.AccessKeyID, val.SecretAccessKey)
+		_, err = c.CoreV1().Secrets(ns).Create(backupS3Secret)
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Begion to backup data cluster %q", clusterFrom.ClusterName))
+		// create backup CRD to process backup
+		backup := fixture.GetBackupCRDWithBR(tcFrom, backupFolder)
+		_, err = cli.PingcapV1alpha1().Backups(ns).Create(backup)
+		framework.ExpectNoError(err)
+
+		// check backup is successed
+		err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+			tmpBackup, err := cli.PingcapV1alpha1().Backups(ns).Get(backup.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			// Check the status in conditions one by one,
+			// if the status other than complete or failed is running
+			for _, condition := range tmpBackup.Status.Conditions {
+				if condition.Type == v1alpha1.BackupComplete {
+					return true, nil
+				} else if condition.Type == v1alpha1.BackupFailed {
+					return false, errors.NewInternalError(nerrors.New(condition.Reason))
+				}
+			}
+			return false, nil
+		})
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Begion to Restore data cluster %q", clusterTo.ClusterName))
+		// create restore CRD to process restore
+		restore := fixture.GetRestoreCRDWithBR(tcTo, backupFolder)
+		_, err = cli.PingcapV1alpha1().Restores(ns).Create(restore)
+		framework.ExpectNoError(err)
+
+		// check restore is successed
+		err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+			tmpRestore, err := cli.PingcapV1alpha1().Restores(ns).Get(restore.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			// Check the status in conditions one by one,
+			// if the status other than complete or failed is running
+			for _, condition := range tmpRestore.Status.Conditions {
+				if condition.Type == v1alpha1.RestoreComplete {
+					return true, nil
+				} else if condition.Type == v1alpha1.RestoreFailed {
+					return false, errors.NewInternalError(nerrors.New(condition.Reason))
+				}
+			}
+			return false, nil
+		})
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Check the correctness of cluster %q and %q", clusterFrom.ClusterName, clusterTo.ClusterName))
+		isSame, err := oa.DataIsTheSameAs(&clusterFrom, &clusterTo)
+		framework.ExpectNoError(err)
+		if !isSame {
+			framework.ExpectNoError(nerrors.New("backup database and restore database is not the same"))
+		}
+
+		// delete backup data in S3
+		err = cli.PingcapV1alpha1().Backups(ns).Delete(backup.Name, &metav1.DeleteOptions{})
+		framework.ExpectNoError(err)
```

Comment:
should we check the backup data is actually been deleted?

Reply:
I think that's needed , I'll add it .

Reply:
done